### PR TITLE
[QNN EP] Op support: LayerNorm, Asin, Sign

### PR DESCRIPTION
**Pull request by HectorSVC**
### Description
Add op support for LayerNorm, Asin, Sign.
Enable QDQ node unit support for Sin Op

### CODE REVIEW

File: onnxruntime/core/optimizer/qdq_transformer/selectors_actions/qdq_selectors.h [hunk 1]
```diff
@@ -139,7 +139,16 @@ class GemmNodeGroupSelector : public NodeGroupSelector {
...
+class InstanceAndLayerNormalizationNodeGroupSelector : public NodeGroupSelector {
+ private:
+  bool Check(const GraphViewer& graph_viewer, const Node& node,
+             const std::vector<const Node*>& dq_nodes,
+             const std::vector<const Node*>& q_nodes) const override;
+};
+
+// Input: DQ nodes for input, scale, and B
+// Output: Q node for output
+class LayerNormalizationNodeGroupSelector : public NodeGroupSelector {
```
Comment (adrianlizarraga) [inline onnxruntime/core/optimizer/qdq_transformer/selectors_actions/qdq_selectors.h:151]: Is this used?

File: onnxruntime/core/providers/qnn/builder/opbuilder/layer_norm_op_builder.cc [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+  const auto& outputs = node_unit.Outputs();
+  ONNX_NAMESPACE::DataType output_data_type = outputs[0].node_arg.Type();
+  ORT_RETURN_IF(!is_quantized_model && output_data_type != float_elem_type, "QNN LayerNorm data type " + *output_data_type + " is not supported in CPU backend.");
+  ORT_RETURN_IF(outputs.size() > 1, "QNN LayerNorm only support 1 output.");
+
+  return AddToModelBuilder(qnn_model_wrapper, node_unit, logger, is_quantized_model, true);
+}
+
+Status LayerNormOpBuilder::ProcessAttributesAndOutputs(QnnModelWrapper& qnn_model_wrapper,
+                                                          const NodeUnit& node_unit,
```
Comment (adrianlizarraga) [inline onnxruntime/core/providers/qnn/builder/opbuilder/layer_norm_op_builder.cc]: formatting is a bit off

File: onnxruntime/core/providers/qnn/builder/opbuilder/layer_norm_op_builder.cc [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+Status LayerNormOpBuilder::IsOpSupported(QnnModelWrapper& qnn_model_wrapper,
+                                            const NodeUnit& node_unit,
+                                            const logging::Logger& logger,
+                                            bool is_quantized_model) const {
+  const auto float_elem_type = ONNX_NAMESPACE::Utils::DataTypeUtils::ToType("float");
+
+  // Check input type is float for CPU.
+  const auto& inputs = node_unit.Inputs();
+  ONNX_NAMESPACE::DataType input_data_type = inputs[0].node_arg.Type();
+  ORT_RETURN_IF(!is_quantized_model && input_data_type != float_elem_type, "QNN LayerNorm data type " + *input_data_type + " is not supported in CPU backend.");
```
Comment (adrianlizarraga) [inline onnxruntime/core/providers/qnn/builder/opbuilder/layer_norm_op_builder.cc:49]:
```suggestion
  ORT_RETURN_IF(!is_quantized_model && input_data_type != float_elem_type, "QNN LayerNorm data type ", input_data_type->c_str(), " is not supported in CPU backend.");
```

The macro is able to concatenate variadic strings. This avoids the need to concatenate strings unless the error actually happens.

File: onnxruntime/core/providers/qnn/builder/opbuilder/layer_norm_op_builder.cc [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+
+  // Also check output type is float for CPU.
+  const auto& outputs = node_unit.Outputs();
+  ONNX_NAMESPACE::DataType output_data_type = outputs[0].node_arg.Type();
+  ORT_RETURN_IF(!is_quantized_model && output_data_type != float_elem_type, "QNN LayerNorm data type " + *output_data_type + " is not supported in CPU backend.");
```
Comment (adrianlizarraga) [inline onnxruntime/core/providers/qnn/builder/opbuilder/layer_norm_op_builder.cc:54]:
```suggestion
  ORT_RETURN_IF(!is_quantized_model && output_data_type != float_elem_type, "QNN LayerNorm data type ", output_data_type->c_str(), " is not supported in CPU backend.");
```

**Review by jywu-msft — APPROVED** [review]

**Review by adrianlizarraga — APPROVED** [review]

**Review by adrianlizarraga — APPROVED** [review]

**Pull request merged**